File: backend/models/Brand.js
import mongoose from "mongoose";

const brandSchema = new mongoose.Schema(
  {
    brandName: {
      type: String,
      required: [true, "Brand name is required"],
      trim: true,
      unique: true,
    },
    brandImage: {
      type: String,
      required: [true, "Brand image is required"],
      trim: true,
    },
    order: {
      type: Number,
      default: 0,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

const Brand = mongoose.model("Brand", brandSchema);

export default Brand;

File: mobile app/lib/models/brand.dart
class Brand {
  const Brand({
    required this.id,
    required this.brandName,
    required this.brandImage,
    this.order = 0,
    this.isActive = true,
  });

  final String id;
  final String brandName;
  final String brandImage;
  final int order;
  final bool isActive;

  factory Brand.fromJson(Map<String, dynamic> json) {
    return Brand(
      id: json['_id']?.toString() ?? '',
      brandName: json['brandName']?.toString() ?? '',
      brandImage: json['brandImage']?.toString() ?? '',
      order: json['order'] is num ? (json['order'] as num).toInt() : 0,
      isActive: json['isActive'] as bool? ?? true,
    );
  }
}
